clippy_lints/src/types.rs

```diff
@@ -1157,6 +1157,28 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for CastPass {
             lint_fn_to_numeric_cast(cx, expr, ex, cast_from, cast_to);
             if let ExprKind::Lit(ref lit) = ex.node {
                 use syntax::ast::{LitIntType, LitKind};
+                if let LitKind::Int(n, _) = lit.node {
+                    if cast_to.is_fp() {
+                        let from_nbits = int_ty_to_nbits(cast_from, cx.tcx);
+                        let to_nbits = if let ty::Float(FloatTy::F32) = cast_to.sty {
+                            32
+                        } else {
+                            64
+                        };
+                        if from_nbits <= to_nbits {
+                            span_lint_and_sugg(
+                                cx,
+                                UNNECESSARY_CAST,
+                                expr.span,
+                                &format!("casting integer literal to {} is unnecessary", cast_to),
+                                "try",
+                                format!("{}_{}", n, cast_to),
+                                Applicability::HasPlaceholders,
```

Comment:
This suggestion looks `MachineApplicable` to me. Do you have any concerns about this suggestion?

Reply:
Yes, it should be MachineApplicable.
I havn't understand that mean of Applicability before.